internal/terraform/context_apply_deferred_test.go

```diff
@@ -1581,11 +1581,62 @@ output "a" {
 			},
 		},
 	}
+
+	readDataSourceTest = deferredActionsTest{
+		configs: map[string]string{
+			"main.tf": `
+data "test" "a" {
+	name       = "deferred_read"
+}
+
+resource "test" "b" {
+	name = data.test.a.name
+}
+
+output "a" {
+	value = data.test.a
+}
+
+output "b" {
+	value = test.b
+}
+		`,
+		},
+		stages: []deferredActionsTestStage{
+			{
+				inputs: map[string]cty.Value{},
+				wantPlanned: map[string]cty.Value{
```

Comment:
I'd expect to see resource.test.b being deferred as well. Do you know why it isn't?

The resource instance gets marked as deferred here: https://github.com/hashicorp/terraform/blob/main/internal/terraform/node_resource_plan_instance.go#L305

Which should be after the provider was called for planning, and I don't see that changing here.

Reply:
`resource.test.b` is deferred, the test case asserts that as well, but the resource is still somehow in the plan. I would assume that an early access introduced in #34967 would be missing, but I'm not 100% sure on this